### PR TITLE
Add an ADR for Pagination

### PR DESCRIPTION
**Pull request by pezholio**
The AP team are preparing to add pagination / filtering to an endpoint.  Ahead of this, I wanted to get everyone’s eyes on a possible approach, so we can make sure we’re all aligned.

### CODE REVIEW

**Review by brownbearsoftware — COMMENTED** [review]
Including the page info in response headers makes me a little bit uneasy but I can see the pragmatic side of this too.

I would suggest that maybe instead of the pageNumber query parameter being absent meaning page 1 it should mean return everything to preserve current behaviour or we'll encounter weird behaviour when trying to roll this out.  We could then change this to mean page 1 once the frontend is fully deployed.

**Review by richpjames — COMMENTED** [review]
Sounds good, I don't really have any concerns but I have some questions to help my understanding: 
 1. Why go for headers over a query string? 
 2. What does returning the next page give the UI (presumably so we don't have to duplicate the `if (currentPage < totalPages) { currentPage + 1 } else { null }` logic? )

**Comment by pezholio** [comment]
@brownbearsoftware 

> I would suggest that maybe instead of the pageNumber query parameter being absent meaning page 1 it should mean return everything to preserve current behaviour or we'll encounter weird behaviour when trying to roll this out. We could then change this to mean page 1 once the frontend is fully deployed.

Great idea!

File: doc/architecture/decisions/0014-paginating-large-data-sets.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+## Decision
+
+We will implement "page-based pagination", starting first with the `/placement-requests/dashboard`
+endpoint. The page query will look as follows:
+
+```
+/placement-requests/dashboard?page=$pageNumber
+```
+
+Where `$pageNumber` is the page number the client wants to return. If the `$pageNumber` is not]
```
[the remark this answers is not in the document]
Comment (pezholio) [reply doc/architecture/decisions/0014-paginating-large-data-sets.md]: I think it makes sense to be 1-indexed. Willl update the ADR

File: doc/architecture/decisions/0014-paginating-large-data-sets.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                type: integer
+              description: The total number of pages
+            X-Pagination-TotalResults:
+              schema:
+                type: integer
+              description: The total number of results
+            X-Pagination-PageSize:
+              schema:
+                type: integer
+              description: The total number of results
```
Comment (edavey) [inline doc/architecture/decisions/0014-paginating-large-data-sets.md:81]: "The number of results per page" might be clearer / more accurate?

File: doc/architecture/decisions/0014-paginating-large-data-sets.md [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        401:
+          $ref: '#/components/responses/401Response'
+        403:
+          $ref: '#/components/responses/403Response'
+        500:
+          $ref: '#/components/responses/500Response'
+```
+
+Again, using the `PagingAndSortingRepository`, we can get this metadata from the response and return
+it alongside the actual data itself, perhaps using a `Pair`, e.g:
```
Comment (edavey) [inline doc/architecture/decisions/0014-paginating-large-data-sets.md:97]: is it worth including a link to some docs on `Pair` for the benefit of Kotlin newbies? 

e.g. https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/

**Review by edavey — APPROVED** [review]
Looks excellent, Stu thks

**Comment by pezholio** [comment]
Closing this in favour of #859, which has the implementation too.

**Pull request closed without merging**